backend/src/main/java/com/team4/backend/service/UserService.java

```diff
@@ -30,9 +31,14 @@ public UserService(UserRepository userRepository, PBKDF2Encoder pbkdf2Encoder, J
         return userRepository.findByEmailAndPasswordAndIsEnabledTrue(authRequestDto.getEmail(), pbkdf2Encoder.encode(authRequestDto.getPassword()))
                 .map(jwtUtil::generateToken)
                 .switchIfEmpty(Mono.error(new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Can't find user with this credentials")));
+    }
 
+    public Mono<User> findByEmail(String email) {
+        return userRepository.findByEmail(email);
     }
 
+    public Mono<Boolean> isEmailTaken(String email) { return userRepository.existsByEmail(email); }
```

Comment:
maybe rename the method to existsByEmail

Reply:
![image](https://user-images.githubusercontent.com/49820008/134608312-817aee32-74a1-4c16-8423-6b8bfa09295b.png)